### DISCUSSION TITLE
Retrieving values and storing them into Typescript variables

### DISCUSSION THREAD

**Discussion by abelardolg** [cyrilletuzi/angular-async-local-storage#21] [Q&A] [answered]
I have coded this example:

`
myVariable: string;
constructor(protected localStorage: AsyncLocalStorage) {
    const user: User = { firstName: 'Henri', lastName: 'Bergson' };

    this.localStorage.setItem('user', user).subscribe(() => {});

    this.localStorage.getItem<User>('user').subscribe((myUser) => {

      this.myVariable = myUser.lastName;

      console.log(myUser.firstName);
    });

    console.log(this.myVariable); // <- undefined

  }`

Why 'this.myVariable' is 'undefined'? I assigned into this variable the 'myUser.lastName' value.

Best regards,

**Comment by cyrilletuzi** [comment, marked as answer]
All operations in this lib are asynchronous, so your code is wrong. You should wait the setter has finished before accessing the value : 

```typescript
this.localStorage.setItem('user', user).subscribe(() => {

  this.localStorage.getItem<User>('user').subscribe((myUser) => {

    // Do something

  });

});`
```

The rest is just Angular, please use stackoverflow for any support question.

**Comment by hellseyfer** [comment]
I'm experimenting the same problem with storagemap, cant assign subscription response to any variable
